infrastructure/cluster/flux-v2/reana/reana-iam-cronjob.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: iam-reana-sync
+  namespace: reana
+spec:
+  schedule: "0 1 * * *" # every day at 1 am 
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: 1
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: OnFailure 
+          containers:
+            - name: iam-sync
+              image: ghcr.io/vre-hub/vre-iam-reana-sync:latest # to be changed to correct version
+    
+              env:
+            
+              # needed to poll the iam service 
+
+              - name: IAM_SERVER
+                value: "https://iam-escape.cloud.cnaf.infn.it"
+              - name: CLIENT_SECRET
+                valueFrom:
+                  secretKeyRef:
+                    name: reana-vre-iam-client
+                    key: client_secret
+              - name: CLIENT_ID
+                valueFrom:
+                  secretKeyRef:
+                    name: reana-vre-iam-client
+                    key: client_id
+
+              # needed to add users 
+              - name: REANA_ADMIN_TOKEN
+                valueFrom:
+                  secretKeyRef:
+                    name: reana-admin-access-token
+                    key: ADMIN_ACCESS_TOKEN
+
+              # needed for correct DB connection - internals of reana 
+              - name: REANA_SECRET_KEY
+                valueFrom:
+                  secretKeyRef:
+                    name: reana-secrets
+                    key: REANA_SECRET_KEY
+              
+              # to connect to DB       
+              - name: REANA_DB_NAME
+                value: "reana"
+              - name: REANA_DB_PORT
+                value: "6600"
+              - name: REANA_DB_HOST
+                value: "dbod-vre.cern.ch"
+              - name: REANA_DB_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    name: reana-db
+                    key: user
+              - name: REANA_DB_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: reana-db
+                    key: password
+           
+              tty: true
+              imagePullPolicy: Always
+              command:
+              - /bin/sh
+              - -c
+              - date; echo Hello from iam-reana-sync container; 
+                flask reana-admin user-list --admin-access-token $REANA_ADMIN_TOKEN; 
```

Comment:
Have you tried this commands outside of the cronjob ?

I'm asking because there is not kind of check/warning/error raise if the reana output fails or runs unexpectedly  ? For example:

 - Does reana raise any kind of exception/error if  you try to add a user (`reana-admin user-create`) that it is already added ?

Once thing that it could be implemented is the following:
 - list all the users (so that we can check in case of needed)
 - run the generate email list
 - check if there are new users added
 - if so, and only if so, run the add users to reana.

But just a suggestion, this can be added on a second interaction.

Reply:
So the command is this inside the python script

`output = subprocess.check_output([f'flask reana-admin user-create --email {i} --admin-access-token {reana_admin_token}'], shell=True, encoding='utf-8')`

The other flask command is just to test if all the variables are there to run `reana-admin` commands in the container. For the moment, as you spotted, the adds the user for each email, and if the user already exists it throws an error and goes to the next one. I would keep your suggestions as a further implementation.